server/controllers/admin/roles.js

```diff
@@ -0,0 +1,223 @@
+const db = require('../../lib/db');
+const q = require('q');
+const _ = require('lodash');
+
+module.exports.list = list;
+module.exports.detail = detail;
+module.exports.create = create;
+module.exports.update = update;
+module.exports.remove = remove;
+module.exports.affectPages = affectPages;
+module.exports.affectToUser = affectToUser;
+module.exports.listForUser = listForUser;
+
+function list(req, res, next) {
+  const sql = `
+    SELECT BUID(uuid) as uuid, label, project_id 
+    FROM role
+    WHERE project_id=?
+  `;
+  db.exec(sql, req.params.project_id)
+    .then((rows) => {
+      res.json(rows);
+    }).catch(next)
+    .done();
+}
+
+
+
+function detail(req, res, next) {
+
+  const sql = `
+   SELECT BUID(uuid) as uuid, label, project_id 
+   FROM role
+   WHERE  uuid = HUID(?) LIMIT 1
```

Comment:
This works, but makes it harder to modify the SQL in the future.  The better way to do this operation is to convert the `req.params.uuid` to a binary array in JS and then do the comparison in SQL.  Like this:
```js
const binaryUuid = db.bid(req.params.uuid);

db.one(sql, binaryUuid)
```

Reply:
Also, you shouldn't need `LIMIT 1` if you are really querying a uuid.   It should be unique, no?